src/compile/data/facet.ts

```diff
@@ -75,24 +64,47 @@ export class FacetNode extends DataFlowNode {
     return this.name;
   }
 
-  private assembleRowColumnData(channel: 'row' | 'column', crossedDataName: string): VgData {
-    const childChannel = channel === 'row' ? 'y' : 'x';
+  private getChildIndependentFieldWithStep() {
+    const childIndependentFieldsWithStep: ChildIndependentFieldsWithStep = {};
 
+    for (const channel of ['x', 'y'] as ScaleChannel[]) {
+      const childScaleComponent = this.childModel.component.scales[channel];
+      if (childScaleComponent && !childScaleComponent.merged) {
+        const type = childScaleComponent.get('type');
+        const range = childScaleComponent.get('range');
+
+        if (hasDiscreteDomain(type) && isVgRangeStep(range)) {
+          const domain = assembleDomain(this.childModel, channel);
```

Comment:
@domoritz -- even though I call this `assembleDomain` during data assemble already, it does not seem to get the final domain -- possibly because the child nodes have not been walked. 

I'm not sure what's the best way to restructure this to work correctly. 

One structural problem is that assembleData still cause side effect to the data component.
Thus, refactor to move all the side effect to parse may help, but I better let you take a look and decide.